cmd/cli/main.go

```diff
@@ -0,0 +1,361 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	v1 "github.com/k8up-io/k8up/v2/api/v1"
+	"github.com/k8up-io/k8up/v2/cli/restore"
+	"github.com/k8up-io/k8up/v2/cmd"
+	"github.com/urfave/cli/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	Command = &cli.Command{
+		Name:        "cli",
+		Description: "CLI commands that can be executed everywhere, currently just restore is supported",
+		Subcommands: []*cli.Command{
+			{
+				Name:   "restore",
+				Action: RunRestore,
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Destination: &restore.Cfg.Snapshot,
+						Required:    true,
+						Name:        "snapshot",
+						Usage:       "Required ; ID of the snapshot `kubectl get snapshots`, set, via cli or via env: ",
+						EnvVars: []string{
+							"SNAPSHOT",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.SecretRef,
+						Required:    true,
+						Name:        "secretRef",
+						Usage:       "Required ; Set secret name from which You want to take S3 credentials, via cli or via env: ",
+						EnvVars: []string{
+							"SECRET_REF",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.S3Endpoint,
+						Required:    true,
+						Name:        "s3endpoint",
+						Usage:       "Required ; Set s3endpoint from which backup will be taken, via cli or via env: ",
+						EnvVars: []string{
+							"S3ENDPOINT",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.S3Bucket,
+						Required:    true,
+						Name:        "s3bucket",
+						Usage:       "Required ; Set s3bucket from which backup will be taken, via cli or via env: ",
+						EnvVars: []string{
+							"S3BUCKET",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.S3SecretRef,
+						Required:    true,
+						Name:        "s3secretRef",
+						Usage:       "Required ; Set secret name, where S3 username & password are stored from which backup will be taken, via cli or via env: ",
+						EnvVars: []string{
+							"S3SECRETREF",
+						},
+					},
+					&cli.StringFlag{
+						Destination: &restore.Cfg.RestoreMethod,
+						Required:    true,
+						Name:        "restoreMethod",
+						DefaultText: "pvc",
+						Usage:       "Required ; Set restore method [ pvc|s3 ], via cli or via env: ",
```

Comment:
Nit: You can use flag actions to enforce this enum: https://github.com/urfave/cli/blob/main/docs/v2/examples/flags.md#flag-actions